agent/pool/pool.go

```diff
@@ -364,7 +364,7 @@ func (p *ConnPool) dial(
 	tlsRPCType RPCType,
 ) (net.Conn, HalfCloser, error) {
 	// Try to dial the conn
-	d := &net.Dialer{LocalAddr: p.SrcAddr, Timeout: p.Timeout}
+	d := &net.Dialer{LocalAddr: p.SrcAddr, Timeout: DefaultDialTimeout}
```

Comment:
This was changed in https://github.com/hashicorp/consul/pull/11500 but I reverted it back to `DefaultDialTimeout` to be consistent with other usages.